client/pages/login.vue

```diff
@@ -1,15 +1,54 @@
 <script setup lang="ts">
-import { object, string } from 'yup'
+import { ForcePasswordChangeDoneEvent } from '~/utils/events'
 
-interface LoginForm {
-  username: string
-  password: string
+const { t } = useI18n()
+
+type State = 'LOGIN' | 'CHANGE_PASSWORD' | 'COMPLETE_OTP'
+
+function stateToIndex(state: State) {
+  switch (state) {
+    case 'LOGIN':
+      return 0
+    case 'CHANGE_PASSWORD':
+      return 1
+    case 'COMPLETE_OTP':
+      return 2
+  }
 }
 
-const { t, m } = useI18n()
-const { login } = useAuth()
+usePageName(() => t('pages.login.title'))
+definePageMeta({
+  layout: 'centercard',
+  middleware: ['auth'],
+})
+const { currentRoute } = useRouter()
+const { isAuthenticated } = storeToRefs(useAuth())
+
+function parseState(value: string): State {
+  try {
+    if (!isAuthenticated.value) return 'LOGIN'
+    return value.toUpperCase().replace('-', '_') as State
+  } catch (e) {
+    return 'LOGIN'
+  }
```

Comment:
what are you catching ?

Reply:
Cast error

Reply:
There are no cast errors in Typescript, casts don't exist at runtime 🐸